flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/hybrid/tiered/TestingTierProducerAgent.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.io.network.partition.hybrid.tiered;
+
+import org.apache.flink.runtime.io.network.buffer.Buffer;
+import org.apache.flink.runtime.io.network.partition.hybrid.tiered.common.TieredStorageSubpartitionId;
+import org.apache.flink.runtime.io.network.partition.hybrid.tiered.tier.TierProducerAgent;
+
+import java.util.function.Supplier;
+
+/** Test implementation for {@link TierProducerAgent}. */
+public class TestingTierProducerAgent implements TierProducerAgent {
+
+    private Supplier<Boolean> tryStartNewSegmentReturnValueSupplier;
+
+    private Supplier<Boolean[]> tryWriteReturnValueSupplier;
+
+    private int numTotalReceivedBuffers;
+
+    private int tryWriteCounter;
+
+    private boolean isClosed;
+
+    public void setTryStartNewSegmentReturnValueSupplier(
+            Supplier<Boolean> tryStartNewSegmentReturnValueSupplier) {
+        this.tryStartNewSegmentReturnValueSupplier = tryStartNewSegmentReturnValueSupplier;
+    }
+
+    public void setTryWriteReturnValueSupplier(Supplier<Boolean[]> tryWriteReturnValueSupplier) {
+        this.tryWriteReturnValueSupplier = tryWriteReturnValueSupplier;
+    }
+
+    public int numTotalReceivedBuffers() {
+        return numTotalReceivedBuffers;
+    }
+
+    public boolean isClosed() {
+        return isClosed;
+    }
+
+    @Override
+    public boolean tryStartNewSegment(TieredStorageSubpartitionId subpartitionId, int segmentId) {
+        return tryStartNewSegmentReturnValueSupplier.get();
+    }
+
+    @Override
+    public boolean tryWrite(TieredStorageSubpartitionId subpartitionId, Buffer finishedBuffer) {
+        boolean isSuccess = tryWriteReturnValueSupplier.get()[(tryWriteCounter++) % 2];
+        if (isSuccess) {
+            numTotalReceivedBuffers++;
+        }
+        return isSuccess;
+    }
+
+    @Override
+    public void close() {
+        this.isClosed = true;
```

Comment:
It is best to use `closeRunnable()` as it will be more versatile. Mock testing class should be flexible and reusable.

Reply:
Fixed.